### ISSUE TITLE
Is v0.4.3 nnom.py support sigmoid?

### ISSUE THREAD

**Issue by songdaw** [majianjia/nnom#183]
Hi,
   My model use sigmoid activation like:
     activation = 'sigmoid'
     output = Activation(activation)(input)
and I use generate_model() in nnom.py to convert model.
I think should add 
     ('activation' in layer.name and layer.get_config()['activation'] == 'sigmoid') 
to function is_shift_layer() .

**Comment by majianjia** [comment]
For sigmoid, please use Sigmoid layer instead of Activation("sigmoid") at the moment

**Comment by songdaw** [comment]
Sorry, I can't find how to use Sigmoid layer, but tf.keras.activations.sigmoid() which produce Lambda layer. Did I miss something?
I use keras=2.6.0.

**Comment by jonnor** [comment]
Is this the right fix, just to add sigmoid to is_shift_layer ? Or are there other complications?

Btw, a workaround might be to use hard_sigmoid, it appears to be implemented

**Comment by songdaw** [comment]
sigmoid is already implemented, try replace act_hard_sigmoid() with act_sigmoid().



---- Replied Message ----
| From | Jon ***@***.***> |
| Date | 03/30/2024 11:20 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [majianjia/nnom] Is v0.4.3 nnom.py support sigmoid? (Issue #183) |

Is this the right fix, just to add sigmoid to is_shift_layer ? Or are there other complications?

Btw, a workaround might be to use hard_sigmoid, it appears to be implemented
